From cdf5b86e04658c3904d203c9f18c77b4a649c54a Mon Sep 17 00:00:00 2001
From: Mario Limonciello <mario.limonciello@dell.com>
Date: Mon, 3 Oct 2016 14:32:47 -0500
Subject: [PATCH] dell: If running in test suite, don't try to get product ID
 from sysfs

It's possible that the test suite was running in an environment without
bind mounted sysfs.
---
 src/fu-provider-dell.c | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/src/fu-provider-dell.c b/src/fu-provider-dell.c
index 598306f..1586b9a 100644
--- a/src/fu-provider-dell.c
+++ b/src/fu-provider-dell.c
@@ -815,15 +815,17 @@ fu_provider_dell_detect_tpm (FuProvider *provider, GError **error)
 						    AS_VERSION_PARSE_FLAG_NONE);
 
 	/* make it clear that the TPM is a discrete device of the product */
-	if (!g_file_get_contents ("/sys/class/dmi/id/product_name",
-				  &product_name,NULL, NULL)) {
-		g_set_error_literal (error,
-				     FWUPD_ERROR,
-				     FWUPD_ERROR_NOT_SUPPORTED,
-				     "Unable to read product information");
-		return FALSE;
+	if (!priv->fake_smbios) {
+		if (!g_file_get_contents ("/sys/class/dmi/id/product_name",
+					  &product_name,NULL, NULL)) {
+			g_set_error_literal (error,
+					     FWUPD_ERROR,
+					     FWUPD_ERROR_NOT_SUPPORTED,
+					     "Unable to read product information");
+			return FALSE;
+		}
+		g_strchomp (product_name);
 	}
-	g_strchomp (product_name);
 	pretty_tpm_name = g_strdup_printf ("%s TPM %s", product_name, tpm_mode);
 	pretty_tpm_name_alt = g_strdup_printf ("%s TPM %s", product_name, tpm_mode_alt);
 
-- 
2.7.4

